src/main/java/org/symphonyoss/symphony/messageml/elements/Element.java

```diff
@@ -383,6 +383,32 @@ void assertParent(Collection<Class<? extends Element>> permittedParents) throws
     }
   }
 
+  /**
+   * Check in above levels if an element has a permitted parent.
+   */
+  void assertParentAtAnyLevel(Collection<Class<? extends Element>> permittedParents) throws InvalidInputException {
+    Element element = this;
+    Boolean permittedParentFound=false;
+    
+    while(!permittedParentFound && element.getParent() != null) {
+      if (permittedParents.contains(element.getParent().getClass())) {
+        permittedParentFound = true;
+      }
+      else {
+        element = element.getParent();
+      }
+    }
+
+    if (!permittedParentFound) {
+      String permittedParentsClassAsString = permittedParents.stream()
+          .map(permittedParentClass -> permittedParentClass.getSimpleName().toLowerCase())
+          .reduce((item, anotherItem) -> String.format("%s, %s", item, anotherItem))
+          .orElse("");
+      throw new InvalidInputException(String.format("Element \"%s\" can only be a child of the following elements: [%s]",
```

Comment:
The assert message here should be different than the one that asserts the direct parent.